### PR TITLE
🌱 make release-notes for pre-releases

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR improve make release-notes target to generate a specific message for pre-releases.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -594,7 +597,11 @@ release-alias-tag: ## Add the release alias tag to the last build tag
 
 .PHONY: release-notes
 release-notes: $(RELEASE_NOTES_DIR) $(RELEASE_NOTES)
-	go run ./hack/tools/release/notes.go --from=$(PREVIOUS_TAG) > $(RELEASE_NOTES_DIR)/$(RELEASE_TAG).md
+	if [ -n "${PRE_RELEASE}" ]; then \
+	echo ":rotating_light: This is a RELEASE CANDIDATE. Use it only for testing purposes. If you find any bugs, file an [issue](https://github.com/kubernetes-sigs/cluster-api-provider/issues/new)." > $(RELEASE_NOTES_DIR)/$(RELEASE_TAG).md; \
```
Comment (CecileRobertMichon) [inline Makefile:601]:
```suggestion
	echo ":rotating_light: This is a RELEASE CANDIDATE. Use it only for testing purposes. If you find any bugs, file an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/new)." > $(RELEASE_NOTES_DIR)/$(RELEASE_TAG).md; \
```

Comment (fabriziopandini) [reply Makefile]: ooops. Fixed!

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

I forgot to make this change in CAPI after adding it to CAPZ, thanks for adding this!

**Pull request merged**